disttask/framework/dispatcher/dispatcher.go

```diff
@@ -470,16 +476,16 @@ func (d *dispatcher) GetAllSchedulerIDs(ctx context.Context, gTaskID int64) ([]s
 	}
 	ids := make([]string, 0, len(schedulerIDs))
 	for _, id := range schedulerIDs {
-		if ok := matchServerInfo(serverInfos, id); ok {
+		if ok := matchServerInfo(serverInfos, id, d.id); ok {
 			ids = append(ids, id)
 		}
 	}
 	return ids, nil
 }
 
-func matchServerInfo(serverInfos map[string]*infosync.ServerInfo, schedulerID string) bool {
+func matchServerInfo(serverInfos map[string]*infosync.ServerInfo, schedulerID string, id string) bool {
```

Comment:
The advantage of using IP + Port is that even if the server is restarted, subtasks can be obtained immediately without waiting for the lease to expire.

If we append a DDL ID, there will be no such benefit. WDYT? @wjhuang2016 @Benjamin2037

Reply:
I will revert it